gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -355,10 +355,25 @@ public void close() throws IOException {
     return new InstrumentedDataWriterDecorator<Object>(writer, this.taskState);
   }
 
+  private void buildWriterIfNotPreset() throws IOException {
+    if (!this.writer.isPresent()) {
+      try {
+        this.writer = Optional.of(this.closer.register(buildWriter()));
+      } catch (SchemaConversionException sce) {
+        throw new IOException("Failed to build writer for fork " + this.index, sce);
+      }
+    }
+  }
+
   /**
    * Get new records off the record queue and process them.
    */
   private void processRecords() throws IOException, DataConversionException {
+
+    if (this.taskState.getPropAsBoolean(ConfigurationKeys.WRITER_CREATE_EMPTY_FILE_KEY)) {
```

Comment:
I suggest defining the default (`false`) explicitly in `ConfigurationKeys` and use the default here.